staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -600,6 +623,7 @@ func interpretStrategicMergePatchError(err error) error {
 	}
 }
 
+// patchToUpdateOptions creates an UpdateOptions with the same field values as the provided PatchOptions.
```

Comment:
interesting... this means admission won't have visibility to the PatchOptions#force option that was set

Reply:
I think I might not have this right.  In `handler.UpdateResource` we have pre-existing code that will convert `UpdateOptions` to `CreateOptions` to allow for put (create or update) semantics. It that particular case it seemed appropriate to provide the options to the admission webhook that matched whatever operation was being performed, and it's a bit less controversial because `UpdateOptions` and `CreateOptions` options are identical (today at least). But being able to admission check `PatchOptions#force` would be valuable?

The main down side to passing in `PatchOptions` instead of `UpdateOptions` or `CreateOptions` would be that admission webhook authors no longer have a 1:1 association between the "operation" and the "operation options".

Reply:
> The main down side to passing in PatchOptions instead of UpdateOptions or CreateOptions would be that admission webhook authors no longer have a 1:1 association between the "operation" and the "operation options".

Yeah, that's not great. One of the nice things about the way patch is wired is that storage and admission only has to think in terms of create and update. I don't like losing the force field from patch options, but I also am not thrilled about letting webhook authors treat patch and update differently. Hmmm.